tests/Doctrine/Tests/DBAL/Functional/Schema/PostgreSqlSchemaManagerTest.php

```diff
@@ -348,6 +348,25 @@ public function testListQuotedTable() : void
         self::assertNull($comparator->diffTable($offlineTable, $onlineTable));
     }
 
+    public function testListTableDetailsWithNumberSchema() : void
+    {
+        $this->connection->exec('CREATE SCHEMA IF NOT EXISTS "001_test"');
+
+        $offlineTable = new Schema\Table('"001_test".test');
+        $offlineTable->addColumn('id', 'integer');
+        $this->schemaManager->dropAndCreateTable($offlineTable);
+
+        $onlineTable = $this->connection->transactional(function () : Table {
+            $this->connection->exec('SET LOCAL search_path = "001_test"');
```

Comment:
The Schema Manager API supports the specification of a database in which the table should be looked up. 

https://github.com/doctrine/dbal/blob/7ff15cbcc12bc7e66e9bc56cf3f3a60b58294ded/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php#L156

Would it make sense to implement the support for this parameter for PostgreSQL instead of having to modify the global state?

Reply:
> and `PostgreSqlPlatform` actually ignores them

Correct. Why not implement the handling of this parameter in DBAL instead of having to use platform-specific SQL?

> '001_test' is schema not database

Correct. In DBAL, `$database` means schema or database depending on the platform. In the case of `listTableColumns()`, a table is looked up within a schema.

Reply:
probably i have some misunderstanding, but i try to explain my point of view:

> Correct. Why not implement the handling of this parameter in DBAL instead of having to use platform-specific SQL?

because this is not happines in real life: https://github.com/doctrine/migrations/blob/b13df49bd2170a6646469081a7e9713650e1bec5/lib/Doctrine/Migrations/Tracking/TableUpdater.php#L84

in this code doctrine migrations try to find definition of same table used when they call 'select * from migrations_version'

in our cases we use multiple schemas (as i can see in DBAL it calls namespaces) for account segregation for one application which means that every schema in one database has same table structure.

Reason why we not use one database per account is simple - for better performance - PG cannot share connection between databases... but this is a topic for another discussion...

as i see in DBAL only 2 platforms support namespaces - PG and SQL server and as it appears only PG has ability to manually specify order used to search table by short name, for example if i ```SET search_path = one, public``` PG try to search tables in this order:
- pg_temp (if current session uses temporary tables)
- pg_catalog (if you not specify it in search_path)
- one
- public

Reply:
I see the use case. I'll let the PR hang for a few days in case if anyone is not okay with it and then will merge. Mind that if the PR targets `master`, it will not be released earlier than in summer.

> because this is not happiness in real life

It's really beyond the scope of the issue. Limitations of a high-level library are not really something that should define the design of the lower-level one. Especially if they belong to the same organization.